app/main.py

```diff
@@ -1,3 +1,47 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, new: int | float | Distance) -> Distance:
+        if isinstance(new, (int, float)):
+            return Distance(self.km + new)
+        elif isinstance(new, Distance):
+            return Distance(self.km + new.km)
```

Comment:
no need to use isinstance twice here, if there will be an unsupported type, it should raise an error